adds session module and boilerplate UI

gabeweaver opened the pull request:
# Why
So we can have a standardized approach towards authentication and
session management amongst the booster kits.

# What
* Replaces `isomorphic-fetch` with `axios` for client-side HTTP
* Adds `session` module with a common reducer and action directories
for `logIn`, `signUp`, `signOut`, and `resetPassword`.
* Adds `universal/feature` directory to hold completely standalone
features that can be loaded up at any route - with matching features in
the `session` module
* Refactors `lib` to incude `api` and a few other things like test and
error helpers.
* Adds some basic form components
* Adds local storage middleware to the redux store
* Adds a few placeholders for global styles and adds resolve alias to
them in the webpack config
* Adds config and global env for selection the server type (phoenix or
rails)…which are wired up to the corresponding services endpoints
configured in `universal/lib/api`

# Clean Up
* I started writing tests for the session module, but upon further
investigation of trying to make them fail purpose without being able to
do so, I decided to hold off until I could pair with someone to
investigate why we’re getting false positives when we shouldn’t be.

bbrock25 commented on config/environments.js:
```diff
@@ -1,13 +1,21 @@
+  production: (config) => ({
+    globals: {
+      ...config.globals,
+      '__SERVER_URL__': string('https://api.foresight.io/api/v1')
```
?

gabeweaver replied:
using this to test stuff...but also adding it to the config so it can easily be swapped out with whatever endpoint we need to hit...

bbrock25 commented on package.json:
```diff
@@ -171,19 +171,21 @@
     "postcss-font-magician": "^1.4.0",
     "postcss-loader": "^1.2.2",
-    "postcss-modules": "^0.6.3",
+    "postcss-modules": "^0.6.4",
```
did these changes address the silly import issues?

gabeweaver replied:
no and i don't think we are technically using this dep...i just do a `yarn upgrade-interactive` on every PR...that CLI command is available in 0.20.0.

I'll remove it another PR that address general dependency clean up.

gabeweaver commented:
@bbrock25 yea I added several tests, but on the module specifically, it was passing even if I changed some of the expected and stuff...which It should have failed but didn't. Not sure if it's something with `nock` or with the action itself (although it works just fine in the real world)...both logIn, signUp, and resetPassword using a thunk wrapped in another promise since redux-form expects a promise to be returned.

bbrock25 commented:
@gabeweaver yeah, the nock stuff can be pretty finicky, i've spent a bit of time getting that setup just right in the past. hopefully we can take 30 sometime this week and hammer it out. i'm sure its something silly
